planex/planex_handler.py

```diff
@@ -0,0 +1,65 @@
+from typing import TextIO
+import urllib.request
+import json
+from datetime import datetime
+from bs4 import BeautifulSoup
+
+
+DUMP_FILE_PLANEX = "planex.json"
+PLANEX_URL = "http://planex.insa-toulouse.fr/"
+# 10 min
+CUSTOM_MESSAGE_INTERVAL = 10 * 60 * 1000
+
+
+def parse_html(html_string: str) -> BeautifulSoup:
+    return BeautifulSoup(html_string, "html.parser")
+
+def is_planex_down() -> bool:
+    try:
+        with urllib.request.urlopen(PLANEX_URL) as response:
+            if response.getcode() != 200:
+                return False
+            else:
+                html = parse_html(response.read())
+                title = html.find("title")
+                # Note that there's no title tag in the HTML when planex is working normally
+                return title != None and title.text == "503 Service Unavailable"
+    except:
+        print("Error processing following url: " + PLANEX_URL)
+        return True
+
+
+def get_json(file: TextIO) -> dict:
+    file_json = {
+        "info": {},
+        "isPlanexDown": None
+    }
+    try:
+        file_json = json.load(file)
+    except json.JSONDecodeError as e:
+        print("Error reading file " + file.name)
+        print(e)
+
+    if not ("info" in file_json):
+        file_json["info"] = {}
+
+    info = file_json["info"]
+    if not ("last_checked" in info) or info[
+            "last_checked"] < datetime.now().timestamp() * 1000 - CUSTOM_MESSAGE_INTERVAL:
+        info["last_checked"] = datetime.now().timestamp() * 1000
+    file_json["isPlanexDown"] = is_planex_down()
+    return file_json
+
+
+def write_json(data: dict, f: TextIO):
+    f.seek(0)
+    f.truncate(0)
+    json.dump(data, f)
+
+
+def main():
+    with open(DUMP_FILE_PLANEX, 'w+', encoding='utf-8') as f:
```

Comment:
I'm getting an error :
```
$ python3 planex_handler.py
Error reading file planex.json
Expecting value: line 1 column 1 (char 0)
```
(This is with python 3.9, not 3.6)

Might be because the file is only open for writing ?

Reply:
Turns out `w+` only reads the new (empty) buffer. Proposed solution would try parsing an empty file. Fixed using `r+` and ensuring the file is created in the shellscript.